test/unit/BPool.t.sol

```diff
@@ -1557,31 +1558,176 @@ contract BPool_Unit_ExitswapPoolAmountIn is BasePoolTest {
+      _fuzz.tokenOutBalance,
+      _fuzz.tokenOutDenorm,
+      _fuzz.totalSupply,
+      _fuzz.totalWeight,
+      _fuzz.poolAmountIn,
+      _fuzz.swapFee
+    );
+
+    vm.expectEmit(true, true, true, true);
+    emit BPool.LOG_EXIT(address(this), tokenOut, _tokenAmountOut);
+
+    bPool.exitswapPoolAmountIn(tokenOut, _fuzz.poolAmountIn, 0);
+  }
+
+  function test_Pull_PoolShare(ExitswapPoolAmountIn_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    uint256 _balanceBefore = bPool.balanceOf(address(this));
+
+    bPool.exitswapPoolAmountIn(tokenOut, _fuzz.poolAmountIn, 0);
+
+    assertEq(bPool.balanceOf(address(this)), _balanceBefore - _fuzz.poolAmountIn);
+  }
+
+  function test_Burn_PoolShare(ExitswapPoolAmountIn_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    uint256 _totalSupplyBefore = bPool.totalSupply();
+    uint256 _exitFee = bmul(_fuzz.poolAmountIn, EXIT_FEE);
+
+    bPool.exitswapPoolAmountIn(tokenOut, _fuzz.poolAmountIn, 0);
+
+    assertEq(bPool.totalSupply(), _totalSupplyBefore - bsub(_fuzz.poolAmountIn, _exitFee));
+  }
+
+  function test_Push_PoolShare(ExitswapPoolAmountIn_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    address _factoryAddress = bPool.call__factory();
+    uint256 _balanceBefore = bPool.balanceOf(_factoryAddress);
+    uint256 _exitFee = bmul(_fuzz.poolAmountIn, EXIT_FEE);
+
+    bPool.exitswapPoolAmountIn(tokenOut, _fuzz.poolAmountIn, 0);
+
+    assertEq(bPool.balanceOf(_factoryAddress), _balanceBefore - _fuzz.poolAmountIn + _exitFee);
+  }
+
+  function test_Push_Underlying(ExitswapPoolAmountIn_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    uint256 tokenAmountOut = calcSingleOutGivenPoolIn(
+      _fuzz.tokenOutBalance,
+      _fuzz.tokenOutDenorm,
+      _fuzz.totalSupply,
+      _fuzz.totalWeight,
+      _fuzz.poolAmountIn,
+      _fuzz.swapFee
+    );
+
+    vm.expectCall(address(tokenOut), abi.encodeWithSelector(IERC20.transfer.selector, address(this), tokenAmountOut));
+    bPool.exitswapPoolAmountIn(tokenOut, _fuzz.poolAmountIn, 0);
+  }
+
+  function test_Returns_TokenAmountOut(ExitswapPoolAmountIn_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    uint256 _expectedTokenAmountOut = calcSingleOutGivenPoolIn(
+      _fuzz.tokenOutBalance,
+      _fuzz.tokenOutDenorm,
+      _fuzz.totalSupply,
+      _fuzz.totalWeight,
+      _fuzz.poolAmountIn,
+      _fuzz.swapFee
+    );
+
+    (uint256 _tokenAmountOut) = bPool.exitswapPoolAmountIn(tokenOut, _fuzz.poolAmountIn, 0);
+
+    assertEq(_tokenAmountOut, _expectedTokenAmountOut);
+  }
+
+  function test_Emit_LogCall(ExitswapPoolAmountIn_FuzzScenario memory _fuzz) public happyPath(_fuzz) {
+    vm.expectEmit(true, true, true, true);
+    bytes memory _data = abi.encodeWithSelector(BPool.exitswapPoolAmountIn.selector, tokenOut, _fuzz.poolAmountIn, 0);
+    emit BPool.LOG_CALL(BPool.exitswapPoolAmountIn.selector, address(this), _data);
+
+    bPool.exitswapPoolAmountIn(tokenOut, _fuzz.poolAmountIn, 0);
```

Comment:
Here I propose the same as in https://github.com/defi-wonderland/balancer-v1-amm/pull/32#discussion_r1616558393 in relation with the input arguments.